### ISSUE TITLE
Infrared support

### ISSUE THREAD

**Issue by shaiu** [tuya/tuya-smart-life#121]
# The problem

Is there an option with this SDK to support Infrared categories?

I have on my smart life app an AC which is controlled by infrared tuya device. I can see the remote on my app but there is no mention about it in the SDK.

I've tried to play around with the SDK and only succeeded to turn on/off with the command {"code":"switch", "value": true/false}
and also {"code":"temp", "value":26} but other commands like "mode", "fan" don't work.

This is the diagnostics I get from HomeAssistant:
`{
  "home_assistant": {
    "installation_type": "Home Assistant Core",
    "version": "2024.3.0.dev0",
    "dev": true,
    "hassio": false,
    "virtualenv": true,
    "python_version": "3.11.7",
    "docker": false,
    "arch": "arm64",
    "timezone": "Asia/Jerusalem",
    "os_name": "Darwin",
    "os_version": "14.3",
    "run_as_root": false
  },
  "custom_components": {},
  "integration_manifest": {
    "domain": "tuya",
    "name": "Tuya",
    "codeowners": [
      "@Tuya",
      "@zlinoliver",
      "@frenck"
    ],
    "config_flow": true,
    "dependencies": [
      "ffmpeg"
    ],
    "dhcp": [
      {
        "macaddress": "105A17*"
      },
      {
        "macaddress": "10D561*"
      },
      {
        "macaddress": "1869D8*"
      },
      {
        "macaddress": "381F8D*"
      },
      {
        "macaddress": "508A06*"
      },
      {
        "macaddress": "68572D*"
      },
      {
        "macaddress": "708976*"
      },
      {
        "macaddress": "7CF666*"
      },
      {
        "macaddress": "84E342*"
      },
      {
        "macaddress": "D4A651*"
      },
      {
        "macaddress": "D81F12*"
      }
    ],
    "documentation": "https://www.home-assistant.io/integrations/tuya",
    "integration_type": "hub",
    "iot_class": "cloud_push",
    "loggers": [
      "tuya_iot"
    ],
    "requirements": [
      "tuya-device-sharing-sdk==0.1.9",
      "segno==1.5.3"
    ],
    "is_built_in": true
  },
  "data": {
    "endpoint": "https://apigw.tuyaeu.com",
    "terminal_id": "1707085806436vrBD5n",
    "mqtt_connected": true,
    "disabled_by": null,
    "disabled_polling": false,
    "id": "bf6e7e369bcb1e76f13lsl",
    "name": "Living room AC",
    "category": "infrared_ac",
    "product_id": "qzktzhehinzsz2je",
    "product_name": "Air Conditioning",
    "online": true,
    "sub": true,
    "time_zone": "+03:00",
    "active_time": "2023-09-09T15:37:22+00:00",
    "create_time": "2023-09-09T15:37:22+00:00",
    "update_time": "2023-09-09T15:37:22+00:00",
    "function": {
      "F": {
        "type": "ENUM",
        "value": {
          "min": 0,
          "max": 3,
          "scale": 0,
          "step": 1,
          "type": "Integer"
        }
      },
      "M": {
        "type": "ENUM",
        "value": {
          "min": 0,
          "max": 4,
          "scale": 0,
          "step": 1,
          "type": "Integer"
        }
      },
      "PowerOff": {
        "type": "STRING",
        "value": "PowerOff"
      },
      "PowerOn": {
        "type": "STRING",
        "value": "PowerOn"
      },
      "T": {
        "type": "ENUM",
        "value": {
          "min": 16,
          "max": 30,
          "scale": 0,
          "step": 1,
          "type": "Integer"
        }
      }
    },
    "status_range": {
      "wind": {
        "type": "ENUM",
        "value": {
          "min": 0,
          "max": 3,
          "scale": 0,
          "step": 1,
          "type": "Integer"
        }
      },
      "mode": {
        "type": "ENUM",
        "value": {
          "min": 0,
          "max": 4,
          "scale": 0,
          "step": 1,
          "type": "Integer"
        }
      },
      "power": {
        "type": "BOOLEAN",
        "value": {}
      },
      "temp": {
        "type": "ENUM",
        "value": {
          "min": 16,
          "max": 30,
          "scale": 0,
          "step": 1,
          "type": "Integer"
        }
      }
    },
    "status": {},
    "home_assistant": {
      "name": "Living room AC",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "remote.living_room_ac",
            "state": "unknown",
            "attributes": {
              "friendly_name": "Living room AC",
              "supported_features": 0
            },
            "last_changed": "2024-02-06T20:45:01.021891+00:00",
            "last_updated": "2024-02-06T20:45:01.021891+00:00"
          }
        }
      ]
    },
    "set_up": true,
    "support_local": true
  }
}`

**Comment by yoavf** [comment]
Same question, and also for RF. I have an Infrared + RF bridge. These are the relevant devices from the diagnostic info:
```json
{
   {
        "id": "bf3bca2b979c8fdbde5pm5",
        "name": "RF blaster",
        "category": "wnykq",
        "product_id": "j2n1pttd32fequga",
        "product_name": "ZC-XZ IR-RF",
        "online": false,
        "sub": false,
        "time_zone": "+02:00",
        "active_time": "2023-02-23T19:35:26+00:00",
        "create_time": "2023-02-23T19:35:26+00:00",
        "update_time": "2023-02-23T19:35:26+00:00",
        "function": {},
        "status_range": {},
        "status": {},
        "home_assistant": {
          "name": "RF blaster",
          "name_by_user": null,
          "disabled": false,
          "disabled_by": null,
          "entities": []
        },
        "set_up": false,
        "support_local": true
      },
      {
        "id": "bf5b9b395f5434943cdmsk",
        "name": "White Light",
        "category": "rf_white_light",
        "product_id": "0000009wke",
        "product_name": "\u767d\u5149\u706f",
        "online": false,
        "sub": true,
        "time_zone": "+02:00",
        "active_time": "2024-02-04T12:33:57+00:00",
        "create_time": "2024-02-04T12:33:57+00:00",
        "update_time": "2024-02-04T12:33:57+00:00",
        "function": {},
        "status_range": {},
        "status": {},
        "home_assistant": {
          "name": "White Light",
          "name_by_user": null,
          "disabled": false,
          "disabled_by": null,
          "entities": []
        },
        "set_up": false,
        "support_local": true
      },
      {
        "id": "bf3e1828db5a623af3cqz2",
        "name": "Fan Light",
        "category": "rf_fan_light",
        "product_id": "0000008sff",
        "product_name": "\u98ce\u6247\u706f",
        "online": false,
        "sub": true,
        "time_zone": "+02:00",
        "active_time": "2024-02-04T12:36:27+00:00",
        "create_time": "2024-02-04T12:36:27+00:00",
        "update_time": "2024-02-04T12:36:27+00:00",
        "function": {},
        "status_range": {},
        "status": {},
        "home_assistant": {
          "name": "Fan Light",
          "name_by_user": null,
          "disabled": false,
          "disabled_by": null,
          "entities": []
        },
        "set_up": false,
        "support_local": true
      },
}
```

**Comment by shihuimiao** [comment]
Not Supported the category of "infrared_ac".

**Comment by shaiu** [comment]
> Not Supported the category of "infrared_ac".

@shihuimiao Yes I know. Is there a way I can contribute and add this support? I will be happy to develop the integration as I'm familiar with APIs and Home assistant.

**Comment by shaiu** [comment]
@shihuimiao any update?

**Comment by shihuimiao** [comment]
You can submit PR according to the development guidelines in Readme.
